tests/library/trace-viewer.spec.ts

```diff
@@ -1439,6 +1439,15 @@ test.skip('should allow showing screenshots instead of snapshots', async ({ runA
   await expect(screenshot).toBeVisible();
 });
 
+test('canvas clipping', async ({ runAndTrace, page, server }) => {
+  const traceViewer = await runAndTrace(async () => {
+    await page.goto(server.PREFIX + '/screenshots/canvas.html');
+    await page.waitForTimeout(1000); // ensure we could take a screenshot
+  });
+
+  await traceViewer.page.pause();
```

Comment:
this test isn't gonna stay, obviously. but you can use it to play around with the feature: `npm run ctest -- --grep 'clipping' --headed`

@dgozman what's the best way of testing this - should we trace a page with a `canvas` and then take a screenshot of the trace, to ensure the screenshot isn't blank? Or somehow compute if it's blank without storing a golden image?

Reply:
I think that would be a flaky test, unfortunately. Perhaps just check some test-specific logging, like we do in recorder tests:

```js
if (isUnderTest) console.error(`drawn 15;23-184;566 screenshot rect on the canvas`);
```

Reply:
thanks, good idea! implemented